controllers/csiaddonsnode_controller.go

```diff
@@ -39,17 +50,69 @@ type CSIAddonsNodeReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CSIAddonsNode object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
 func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	// Fetch CSIAddonsNode instance
+	csiAddonsNode := &csiaddonsv1alpha1.CSIAddonsNode{}
+	if err := r.Client.Get(ctx, req.NamespacedName, csiAddonsNode); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			logger.Info("CSIAddonsNode resource not found")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// TODO: add validation for csiaddonsnode spec
+	nodeID := csiAddonsNode.Spec.Driver.NodeID
+	driverName := csiAddonsNode.Spec.Driver.Name
+	endPoint := csiAddonsNode.Spec.Driver.EndPoint
+	key := r.createKey(csiAddonsNode)
+	logger = logger.WithValues("NodeID", nodeID, "EndPoint", endPoint, "Driver Name", driverName)
+
+	if !csiAddonsNode.DeletionTimestamp.IsZero() {
+		// if deletion timestap is set, the CSIAddonsNode is getting deleted,
+		// delete connections and remove finalizer.
+		logger.Info("Deleting connection")
+		r.ConnPool.Delete(key)
+		if err := r.removeFinalizer(ctx, &logger, csiAddonsNode); err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
```

Comment:
as we are not logging anything err check is required here? just return `ctrl.Result{},err`?